Update gQSPsim with contributions from MathWorks

simkaryote opened the pull request:
THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY.

Commit:
fix error with new session

Commit:
fix warning dialog; user has to make sure to select one cell and right-click for ApplyToAll

Commit:
added method to AddRemoveTable to apply invalid entry styling to table and replaced optim table to call the new method

Commit:
replaced all calls to QSP.makeInvalidStyle in OptimizationPane to addInvalidStyle instead

Commit:
replaced all calls to QSP.makeInvalidStyle in CohortGenerationPane to addInvalidStyle instead

Commit:
replaced all calls to QSP.makeInvalidStyle in GlobalSensitivityAnalysisPane to addInvalidStyle instead

Commit:
replaced all calls to QSP.makeInvalidStyle in VirtualPopulationGenerationPane to addInvalidStyle instead

Commit:
replaced all calls to QSP.makeInvalidStyle in SimulationPane to addInvalidStyle instead

Commit:
add if statement line 1131 to avoid error when Data is empty

Commit:
fix issue #63 - changing values in filtered vis param table will now reflect properly in the original source table

Commit:
fixed issue #77

Commit:
fix issue #73 - call redrar for Group and ID dropdown in the callback for dataset dropdown

Commit:
fix issue #75 - edge case error that occurs only when 1 task is configured for optimization - included conditional statement to take care of the edge case

Commit:
fix issue #71

Commit:
fix issue #52 - enable delete, restore multiple nodes

Commit:
fix bugs with restoring nodes

Commit:
enable duplication of multiple selected nodes

Commit:
fix issue #81 - no warning now if plugin directory is same as root directory

Commit:
fix issue #85 #82

Commit:
fix issue #84

Commit:
fix issue #83

Commit:
fix issue #89 - make the uifigure modal only >=R2020b

Commit:
fix issues #47, #91 - changed architecture such that logger files are always created in the session root directory with the current name of session - logger file names and location change everytime the session name and root directory are changed respectively

Commit:
fix issue #92 - this was a bug in the main app. backend changed current selectedsession node to the most recent created session while the user selected a different session in frontend

Commit:
fix #97

Commit:
fix #95

Commit:
add note taking ability for logger

Commit:
add menu to launch model manager

faugusti committed:
fix table selection bug

faugusti committed:
merge changes from dev branch

Commit:
added placeholder to have summary views for functionality top nodes

Commit:
Increased size of Summary label and made table sortable

feigelman committed:
Set root directory by default for model manager

feigelman committed:
allow one instance of model manager

feigelman committed:
Merge pull request #98 from gQSPSim/kmahalin/modelmanagerlaunchmenu

add menu to launch model manager

Commit:
addinputs for absolute and relative tolerance in task edit view

simkaryote committed:
Merge pull request #80 from gQSPSim/kmahalin/issues_75_73

Kmahalin/issues 75 73

simkaryote committed:
Merge pull request #101 from gQSPSim/gsa

Gsa

simkaryote committed:
Merge pull request #88 from gQSPSim/kmahalin/issue52

Kmahalin/issue52

simkaryote committed:
Merge pull request #104 from gQSPSim/kmahalin/issue103

add inputs for absolute and relative tolerance in task edit view

Commit:
add constraints for absolute and relative tolerance

Commit:
add enable/disable parallel button

Commit:
synchronize toplevel session object with node level so changes in parallel status refelect everywhere

Commit:
fix issue #107 - extend parallel button for all functionality nodes

Commit:
add dependency icons

Commit:
add git button in session edit view

Commit:
add git button to functionality nodes

Commit:
fix issue #111 - include folder structure inputs for virtual subjects in table and dataset selection in simulation edit view

Commit:
fix issue #100 - add columns specified in confluence notes, add TimeOfCreation property to functionality nodes

Commit:
fix title issue for simulation edit view

Commit:
fix issue with write to csv. the paths were not getting updated with session and logger object

Commit:
adding FunctionalitySummaryPane class

Commit:
adding dependency icon

Commit:
toggle the logic for saving checkpoints

simkaryote committed:
Merge pull request #106 from gQSPSim/kmahalin/issue103

add constraints for absolute and relative tolerance

Commit:
add timer callback logic

simkaryote committed:
Merge pull request #123 from gQSPSim/kmahalin/issue49_taskpropssave

Kmahalin/issue49 taskpropssave

simkaryote committed:
Merge pull request #121 from gQSPSim/kmahalin/task133_applyvaluetoall

Kmahalin/task133 applyvaluetoall

simkaryote committed:
Merge pull request #114 from gQSPSim/kmahalin/issue113_gitbutton

Kmahalin/issue113 gitbutton

simkaryote committed:
Merge branch 'dev' into kmahalin/issue107_parallelbutton

simkaryote committed:
Merge pull request #112 from gQSPSim/kmahalin/issue107_parallelbutton

Kmahalin/issue107 parallelbutton

simkaryote committed:
Workaround for a bug in uitable.

simkaryote committed:
Remove left over commented out code.

simkaryote committed:
Remove UI elements from load operation.

simkaryote committed:
Testing failure job

simkaryote committed:
Validated that a failure in the test fails the job in github. Undoing he force failure that was added to test this.

simkaryote committed:
Merge branch 'dev' into kmahalin/FunctionalitySummaryView

simkaryote committed:
Fixed faulty getSummaryTableItems.

simkaryote committed:
Merge pull request #166 from gQSPSim/pr_115

Fix to Karthiga's pull request #115

simkaryote committed:
Fixing graphics issues with OptimizationPlots

simkaryote committed:
Start reconciling offline work with new/fixed pull requests.

simkaryote committed:
Merge branch 'dev' of github.com:gQSPSim/gQSPSim into dev

simkaryote committed:
Merge branch 'dev' into dev_pax

simkaryote committed:
Added in Karthiga's 116

simkaryote committed:
Merge in Karthiga's 117

simkaryote committed:
Merge Karthiga's 118

simkaryote committed:
Merge Karthiga's 119

simkaryote committed:
Merge Karthiga's 120

simkaryote committed:
Merge Karthiga's 122

simkaryote committed:
Don't care about tracking changes in Sessions directory.

simkaryote committed:
Resetting baseline

simkaryote committed:
Leave a merge issue pending. Commenting out code to get this to run again.

simkaryote committed:
Merge cleanup

simkaryote committed:
Preparing to move Panes for refactor.

simkaryote committed:
New AppContainer and ViewPane init.

simkaryote committed:
Proper initialization of FigurePanel.
Added resizing of the tree control.

simkaryote committed:
Proper forwarding of baseclass construction.
Albeit using the new namedargs2cell function..

simkaryote committed:
Move UI main view out of ApplicationUI

simkaryote committed:
Command line test based on API example.

simkaryote committed:
Remove plotFcn from optimization to enable CLI tests

simkaryote committed:
AppContainer based main view. Can't ship this
at the moment.

simkaryote committed:
New layout arrangement for the main view.
This replaces the FlexGrid implementation.

simkaryote committed:
Ported over PaneManager
Mark old fields for removal

simkaryote committed:
Updates for reorg.
1. Update constructor signature.
2. temporary properties while moving to new architecture.

simkaryote committed:
Reorg of layout.

simkaryote committed:
New Pane classes.
Central management of the Panes.
Single toolbar; no button replication per Pane.

simkaryote committed:
Removal of toolbar. Moved to new class.

simkaryote committed:
Regularize names.
Enables common construct and update of all panes.

simkaryote committed:
Communicate alerts via events.

simkaryote committed:
Move PaneManager into OuterShell.

simkaryote committed:
Use Alert event.

simkaryote committed:
Enable running in from new PaneToolbar.

simkaryote committed:
Add Summary Pane back.

simkaryote committed:
Turn off addUDF for now. Needs to be cleaned up.

simkaryote committed:
validateRulesAndReactions is a v1.* thing.

simkaryote committed:
More todo.. Update these constructors.

simkaryote committed:
Enable testing on the reorg_pax branch.

simkaryote committed:
Forgot to make the file executable.

simkaryote committed:
Need to enable submodule cloning bc of advLogger.

simkaryote committed:
Update to units. Use nano prefix rather than 10^-9.

simkaryote committed:
Register units in the test setup method. We may want a more global place.

simkaryote committed:
Added modes to ViewPane.
Now includes a Visualization mode.
Would be nice to split the ViewPane into objects for the various modes.

simkaryote committed:
Listen to toolbar notifications.

simkaryote committed:
Toolbar now has events for all the button actions.

simkaryote committed:
Run tests in non-strict mode

simkaryote committed:
Merge branch 'reorg_pax' of insidelabs-git.mathworks.com:compbio/gqspsim_ui into reorg_pax

simkaryote committed:
Add my testing session file. Remove before merge request.

simkaryote committed:
Put 19ab back into testing.

simkaryote committed:
Turn on more releases going back to 18a

simkaryote committed:
Remove old code and adopt new class: panemanager

simkaryote committed:
Will likely need an event to propagate data change.
Ideally this is typed but leave untyped for now.

simkaryote committed:
Activate summary panels

simkaryote committed:
try "drawnow limitrate" for sliding pane.

simkaryote committed:
separate construct from show/update

simkaryote committed:
Added toolbar mode.

simkaryote committed:
Enabled visualization.
Still have issues with the various mechanisms for pane update.

simkaryote committed:
Added enumeration for toolbar mode

simkaryote committed:
Removal of Focus property and cleanup of refocus
method.

simkaryote committed:
Merge branch 'reorg_pax' of insidelabs-git.mathworks.com:compbio/gqspsim_ui into reorg_pax

simkaryote committed:
New Session initialization cleanup
Moving more UI functionality into mainView and out of controller.

simkaryote committed:
Renamed to NewSessionEventData

simkaryote committed:
Menu items are now functional.
Removal of dead code.

simkaryote committed:
Enable the construction of the controller without a view

simkaryote committed:
New IsDirty implementation for MVC

simkaryote committed:
Session nodes update based on dirty/clean messages.

simkaryote committed:
Dirty state connected to close and save if needed.

simkaryote committed:
Saving enabled.
Various bug fixes and old code removal.

simkaryote committed:
Enabled delete and restore.
Close a pane if the corresponding session is closed.

simkaryote committed:
Fix the constructor signature to adhere to Pane
standards

simkaryote committed:
Fixed ModelManager
Added Preferences

simkaryote committed:
OpenRecent enabled.

simkaryote committed:
PluginManager uses access method to get ItemTypes

simkaryote committed:
clean mlint warnings.

simkaryote committed:
Arrange private methods

simkaryote committed:
Cleanup debugging messages.

simkaryote committed:
Delete old refactor code.

simkaryote committed:
Rename OuterShell_UIFigureBased to MainView
This is the top level view of the application.

simkaryote committed:
Rename ApplicationUI to Controller.

simkaryote committed:
Adding 22a and b to the test list.

simkaryote committed:
Revert change that specified the size.

simkaryote committed:
Apply standard names to events.

simkaryote committed:
Formatting.

simkaryote committed:
Adding simple test for QSP Add Item menus

simkaryote committed:
Crude test for Add Item.

simkaryote committed:
Turn back off. Not ready.

simkaryote committed:
Unreported bug fix. If there were no Parameters
for this optimization (e.g. an empty Optimization)
this code would error.

simkaryote committed:
About message moved to controller and
it now updates with the central version number.

Also added the beginning of Folders with an entirely new
architecture. This new one is a fraction of the code previously
used. It introduces a typed folder identified by a tag of the form
<type>:Folder

simkaryote committed:
First move to reuse the FunctionalitySummaryPane
as a generic summary pane.

simkaryote committed:
Generalization to addItem in preparation of Folders
Some cleanup of the code. For instance createTreeNode now
must take a Tag. Tags are the mechanism for the View to find
elements in the tree.

simkaryote committed:
closeActivePane needed to be public.
Generalizations for Folders.

simkaryote committed:
Toolbar initializes in minimized mode.

simkaryote committed:
Fixed mlint warnings.

simkaryote committed:
Cleanup of mlint warnings.
Removal of more dead code.

simkaryote committed:
Addition of context menus.

simkaryote committed:
Minor tidying up.

simkaryote committed:
Update test setup.

simkaryote committed:
Turn on tCLI for MW testing as a prep for github.

simkaryote committed:
Turn on basic UI testing.

simkaryote committed:
Abstract the location of the test project.
This is an ugly workaround to the RootDirectory mishandling in the app.

simkaryote committed:
Expose addSession & loadSession in the public api.

simkaryote committed:
New form of the constructor.
Cleanup cruft from merge with Kathiga's code.

simkaryote committed:
Minor formatting.

simkaryote committed:
run tsimple only.

simkaryote committed:
Updating license to MIT as discussed with
Genentech (Roche)

simkaryote committed:
Updating README for open source release.

simkaryote committed:
turn off preference loading.
a bug with it is preventing tests from running.

simkaryote committed:
Menu accelerators do not support string.
This is true for <= 19b.  in 20a we are OK.

simkaryote committed:
Trying to accomodate 19b.

simkaryote committed:
Cleanup how we launch tests.

simkaryote committed:
Moved the driver to the top of sandbox
to simplify test launching.

simkaryote committed:
Remove MathWorks copyright and author information
The MathWorks' copyright is in the license.txt file.

simkaryote committed:
move towards argument blocks from the older
functional form.

simkaryote committed:
Settings is set on all functionality items.
This may not be what we really want but
lots of existing code depend on it.

simkaryote committed:
Tweak pane layout

simkaryote committed:
Cleanup getSelectionNode.
Still relies on knowledge of the Views
layout/hierarchy but its a lot cleaner than before.

simkaryote committed:
Moved the variant of this widget handling
the various versions of MATLAB and support
for modal dialogs.

simkaryote committed:
getSelectionNode moved to baseclass.
Avoids duplicate code in every derived class.

simkaryote committed:
ParentApp is private again since getSelectedNode
is a local (i.e. baseclass) method now.

simkaryote committed:
Enable notifications for toolbar state buttons.

simkaryote committed:
PaneManager does not construct a PaneToolbar
Rather the MainView does this and connects
the two where appropriate.
Motivation came from the need for the toolbar
to communicate with other parts of the appliction, e.g.
the controller.

There is an open issue here, are the state buttons controlling a state for a session or the entire application or even for a particular functionality?

simkaryote committed:
Turn off 18a,b and 19a. These are not supported
in version 2

simkaryote committed:
Minor formatting.

simkaryote committed:
Adding some TestTags

simkaryote committed:
Updating the test driver

simkaryote committed:
Adding submodule checkout due to dependency
with advanced logger.

simkaryote committed:
UI tests using the uitest framework.

simkaryote committed:
New test driver for use at MathWorks.
This runs UI tests using gitlab-runner.

simkaryote committed:
MATLAB launcher for gitlab testing.

simkaryote committed:
Fixed tag name.

simkaryote committed:
Must run DefinePaths to initialize gQSPsim

simkaryote committed:
What directory is the test launched from?

simkaryote committed:
sorting out a good way to launch tests.

simkaryote committed:
Testing artifacts.

simkaryote committed:
Fix typo

simkaryote committed:
Tests for NoUI

simkaryote committed:
Fix typo.

simkaryote committed:
formatting.

simkaryote committed:
Layout cleanup.

simkaryote committed:
Name changes in a pane are propagated to the tree
browser in the MainView.

simkaryote committed:
Cobertura not available in our gitlab runner.

simkaryote committed:
Enable/disable exploration buttons.
Enable the Settings button on the toolbar.

simkaryote committed:
Enable saving preferences.
Migrate old preferences to new format.
Initialize the View with the recently opened sessions.

simkaryote committed:
Allow access to tests.

simkaryote committed:
Try to concatenate in the other direction.

simkaryote committed:
Enable permanent delete.

simkaryote committed:
Access to ItemTypes from the Controller changed.

simkaryote committed:
Empty Deleted Items enabled.

simkaryote committed:
Enable duplicate items from Treenodes.

simkaryote committed:
ContextMenu cleanup and enable duplicate.

simkaryote committed:
Formatting and access to UIFigure property from Controller.

simkaryote committed:
Add test for duplicate node.

simkaryote committed:
switch over to using -batch v. -r

simkaryote committed:
Dropping 19b since UI and testing frameworks
don't support what we use in QSPSim v.2.0

simkaryote committed:
Cleanup and adding tests

simkaryote committed:
some test use more modern functionalty.
Using testTags to separate those.

simkaryote committed:
Set to the earliest version that will work with the new
test infrastructure.

simkaryote committed:
Select modernTestInfrastructure

simkaryote committed:
Suppress accepted warnings to avoid large logs.
Note the warnings are accepted by the authors of the Test Cases.

simkaryote committed:
Cleanup of constructor signatures.

simkaryote committed:
Removing obsolete TODOs

simkaryote committed:
Delete old attempt at newer UI.
These components cannot be used outside MW.

simkaryote committed:
Handle deleted node context menu.

simkaryote committed:
Moving to submodule for GUI_Layout_Toolbox rather than distributing within gQSPsim.

simkaryote committed:
Resolve merge issues between v1.1 and new code for version 2.0

simkaryote committed:
Remove GUI_Layout_Toolbox_2 as a submodule. It is not ready to be a public repository on GitHub.

simkaryote committed:
Revert to previous BSD-3 license. After more discussion it was determined that we did not need to change the license to MIT after all.

simkaryote committed:
Fix merge of Session.m

A function added to the old v1.1 dev branch was missing from the new update_old_dev branch. This restores it.

simkaryote committed:
register units used by tests.

simkaryote committed:
Fixes test selection based on the MATLAB version.

simkaryote committed:
Run the CLI tests for debugging.

simkaryote committed:
SaveInvalid updated to {all, valid}

SaveInvalid has a lost commit or was incorrectly merged. It appears that the new valid state for SaveInvalid is the set {all, valid}.

simkaryote committed:
Disable one testpoint until it is fixed.

simkaryote committed:
Clean up tests.

simkaryote committed:
Debugging RootDirectory issues with CLI tests.

simkaryote committed:
This remains a TODO.

simkaryote committed:
Fix bug in testing related to RootDirectory

feigelman reviewed (commented):
removing previous comment

feigelman commented:
> running the new version of the code locally after merging I immediately encounter problems with the advanced logger. That directory app/+QSPViewerNew/+Widgets/AdvLogger is empty. I assume that is supposed to track https://github.com/mathworks/advanced-logger as a submodule. However, even after cloning that in I encounter the following error. Do you have a modified local version of this module? It's not in the repository that I can tell. Otherwise I don't know why it's looking for undefined properties in the Logger class.
> 
> No public property 'FileID' for class 'Logger'.
> 
> Error in QSPViewerNew.Widgets.Logger/fcloseLogFile (line 198) if obj.FileID >= 0
> 
> Error in QSPViewerNew.Widgets.Logger/moveLogFile (line 93) obj.fcloseLogFile();
> 
> Error in QSP.Session/updateLoggerName (line 847) moveLogFile(loggerObj, obj.RootDirectory); % moves log file if not present in root dir
> 
> Error in QSP.Session/setSessionName (line 544) updateLoggerName(obj, SessionName)
> 
> Error in QSPViewer.App/refresh (line 81) setSessionName(obj.Session(idx),ThisRawName);
> 
> Error in uix.abstract.AppWithSessionFiles/createUntitledSession (line 223) obj.refresh(); %Call refresh of the main app
> 
> Error in uix.abstract.AppWithSessionFiles/onNew (line 153) createUntitledSession(obj);
> 
> Error in uix.abstract.AppWithSessionFiles>@(h,e)onNew(obj) (line 66) 'Callback',@(h,e)onNew(obj));
> 
> Error while evaluating Menu Callback.

With this error, I cannot launch the app

simkaryote commented:
See

https://stackoverflow.com/questions/3796927/how-do-i-git-clone-a-repo-including-its-submodules

Pax
________________________________
From: feigelman ***@***.***>
Sent: Friday, December 9, 2022 8:31:52 PM
To: gQSPSim/gQSPSim_public ***@***.***>
Cc: Rick Paxson ***@***.***>; Author ***@***.***>
Subject: Re: [gQSPSim/gQSPSim_public] Update gQSPsim with contributions from MathWorks (PR #2)


running the new version of the code locally after merging I immediately encounter problems with the advanced logger. That directory app/+QSPViewerNew/+Widgets/AdvLogger is empty. I assume that is supposed to track https://github.com/mathworks/advanced-logger<https://github.com/mathworks/advanced-logger> as a submodule. However, even after cloning that in I encounter the following error. Do you have a modified local version of this module? It's not in the repository that I can tell. Otherwise I don't know why it's looking for undefined properties in the Logger class.

No public property 'FileID' for class 'Logger'.

Error in QSPViewerNew.Widgets.Logger/fcloseLogFile (line 198) if obj.FileID >= 0

Error in QSPViewerNew.Widgets.Logger/moveLogFile (line 93) obj.fcloseLogFile();

Error in QSP.Session/updateLoggerName (line 847) moveLogFile(loggerObj, obj.RootDirectory); % moves log file if not present in root dir

Error in QSP.Session/setSessionName (line 544) updateLoggerName(obj, SessionName)

Error in QSPViewer.App/refresh<http://QSPViewer.App/refresh> (line 81) setSessionName(obj.Session(idx),ThisRawName);

Error in uix.abstract.AppWithSessionFiles/createUntitledSession (line 223) obj.refresh(); %Call refresh of the main app

Error in uix.abstract.AppWithSessionFiles/onNew (line 153) createUntitledSession(obj);

Error in uix.abstract.AppWithSessionFiles>@(h,e)onNew(obj) (line 66) 'Callback',@(h,e)onNew(obj));

Error while evaluating Menu Callback.

With this error, I cannot launch the app

—
Reply to this email directly, view it on GitHub<https://github.com/gQSPSim/gQSPSim_public/pull/2#issuecomment-1344966707>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AEW6B5IWNC3F2RLGX5BPZNDWMPMQRANCNFSM6AAAAAASDZWKCI>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

simkaryote committed:
Can't directly access the UIFigure. This left accidentally while permissions were still open. Fix bug #8

simkaryote committed:
Fix #11

simkaryote committed:
Contextmenu actions enabled.

simkaryote committed:
Made room for labels.
In general the view panes need some work wrt to layout. Adding an issue to track that.
This addresses Issue #12 in the meantime.

feigelman commented on app/+QSP/+abstract/BaseProps.m:
```diff
@@ -77,6 +72,9 @@
             
         end % constructor       
         
+        function Log(obj, msg)
```
not sure we need Log anymore since we have the Logger class. This probably needs to be removed.

feigelman commented on app/+QSP/+abstract/BasicBaseProps.m:
```diff
@@ -61,6 +58,8 @@
             % Assign PV pairs to properties
             obj.assignPVPairs(varargin{:});
             
+            obj.Version = 2.0;
```
Do we want to externalize version numbering? Otherwise if there are other versions across the code base we will have to maintain consistency for updates.

feigelman commented on app/+QSP/@CohortGeneration/CohortGeneration.m:
```diff
@@ -41,7 +34,7 @@
         GroupName = ''
         Method = 'Distribution' 
-        SaveInvalid = 'Save all virtual subjects' % 'all' = 'Save all virtual subjects' or 'valid' = 'Save valid virtual subjects'
+        SaveInvalid = 'all' % = 'Save all virtual subjects' or 'valid' = 'Save valid virtual subjects'
```
I think ultimately we wanted to use the full strings in the UI. It doesn't matter if that is represented as a shortened string in the backend.

feigelman commented on app/+QSP/@CohortGeneration/CohortGeneration.m:
```diff
@@ -509,7 +598,9 @@ function clearData(obj)
                 obj.SimResults = {};
                 obj.SimFlag = [];
                 
+                obj.Log(['running virtual cohort generation ' obj.Name])
```
probably need to remove this since we are using a new logger

simkaryote replied:
Yes, but that should be a project independently done from the refactor. It is best to try to keep changes atomic for various reasons, one of which is enabling the possibility of reverting or editing a particular change(s).

feigelman commented on app/+QSP/@CohortGeneration/CohortGeneration.m:
```diff
@@ -527,6 +618,12 @@ function clearData(obj)
                     validate(vpopObj,false);
+                    
+                    % add entry to the database
+                    if obj.Session.UseSQL
```
will need to revisit the whole SQL database aspect

feigelman commented on app/+QSP/@Optimization/Optimization.m:
```diff
@@ -640,11 +709,25 @@ function clearData(obj)
                 
                 % Run helper
-                [StatusOK,Message,ResultsFileNames,VPopNames] = optimizationRunHelper(obj);
+                obj.Log(['running optimization ' obj.Name])
```
switch to new logger

simkaryote replied:
sure.. fit with above comment.

feigelman commented on app/+QSP/@Optimization/Optimization.m:
```diff
@@ -640,11 +709,25 @@ function clearData(obj)
                 obj.VPopName = VPopNames;
                 
+                % add entry to the database
+                if obj.Session.UseSQL
```
revisit SQL

feigelman commented on app/+QSP/@Settings/refreshData.m:
```diff
@@ -1,110 +1,134 @@
+    StatusOk = true;
+    Message = '';
+    
+    hWaitBar = waitbar(0, 'Loading session', '', true);    
```
seems like this will work for now. in the future let's move all the waitbars into the viewer?

simkaryote replied:
yes, all UI elements (including waitbars should be in the viewer).

feigelman commented on app/+QSP/@Settings/refreshData.m:
```diff
@@ -1,110 +1,134 @@
+        
+        waitStatus = waitbar(index/numel(obj.Task), hWaitBar, 'Loading Tasks');
+        
+        if ~waitStatus % cancelled
```
same here

feigelman commented on app/+QSP/@Simulation/Simulation.m:
```diff
@@ -309,9 +318,15 @@ function clearData(obj)
                 % Run helper
-                [ThisStatusOK, thisMessage, ResultFileNames, Cancelled, resultsArray] = simulationRunHelper(obj);
-
+                obj.Log(['running simulation ' obj.Name])
```
update to Logger

feigelman commented on app/+QSP/@Simulation/Simulation.m:
```diff
@@ -328,6 +343,10 @@ function clearData(obj)
                 end
                 
+                % add entry to the database
+                if obj.Session.UseSQL
```
revisit

feigelman commented on app/+QSP/@VirtualPopulationGeneration/VirtualPopulationGeneration.m:
```diff
@@ -501,11 +508,19 @@ function clearData(obj)
                 obj.SimResults = {};
                 obj.SimFlag = [];
+                
+                obj.Log(['running vpop generation ' obj.Name])
```
update

feigelman commented on app/+QSP/@VirtualPopulationGeneration/VirtualPopulationGeneration.m:
```diff
@@ -520,6 +535,12 @@ function clearData(obj)
                     validate(vpopObj,false);
+                    
+                    % add entry to the database
+                    if obj.Session.UseSQL
```
revisit

feigelman commented on app/+QSP/plotVirtualCohortGeneration.m:
```diff
@@ -632,7 +627,8 @@
     overlays = [ublb_lines; acc_lines; rej_lines];
     if ~isempty(overlays)
-        uistack(overlays,'top');
+        %uistack is not supported using uiaxes. 
```
need to revisit

feigelman commented on app/+QSPViewer/@App/App.m:
```diff
@@ -519,6 +533,8 @@ function onAddItem(obj,ItemType)
                 error('Invalid tree parent');
             end
             
+            ThisSession.Log(sprintf('added %s tree node item for %s', ItemType, NewName))
```
update

feigelman commented on app/+QSPViewer/@App/App.m:
```diff
@@ -607,6 +623,8 @@ function onRemoveItem(obj)
                 % Change context menu
                 SelNode(nodeIdx).UIContextMenu = obj.h.TreeMenu.Leaf.Deleted;
+                
+                ThisSession.Log(sprintf('Removed item %s', SelNode.Name))
```
update

feigelman commented on app/+QSPViewer/@App/loadSessionFromFile.m:
```diff
@@ -67,7 +60,7 @@
     % convert rules and reactions
     
-    Session.validateRulesAndReactions();
+    %Session.validateRulesAndReactions();
```
why was this removed?

feigelman commented on app/+QSPViewer/@Optimization/Optimization.m:
```diff
@@ -117,8 +108,9 @@
                     UniqueSourceData{index} = cell(0,2);
-                    hDlg = errordlg(Message,'Parameter Import Failed','modal');
-                    uiwait(hDlg);
+                    warning('Parameter Import Failed for %s', UniqueSourceNames{index})
```
why just a warning?

feigelman commented on app/+QSPViewer/@Session/create.m:
```diff
@@ -103,13 +95,127 @@ function create(vObj)
+    'Padding',5,...
+    'Spacing',10);
+
+vObj.h.UseLogFile = uicontrol(...
```
we need to clean up the logger section. Since we're using the advanced logger, don't need these properties anymore?

simkaryote replied:
Probably not.. again that is another project unrelated to the refactor.

feigelman commented on app/+QSPViewer/@Session/create.m:
```diff
@@ -103,13 +95,127 @@ function create(vObj)
+
+%% sqlite database
+
+vObj.h.SQLLayout =  uix.HBox(...
```
revisit if we are leaving out SQL capability

simkaryote replied:
that would be your call. My suggestion would be to leave the code and disable it via some feature control mechanism. Simplest form of that would be to add a table of boolean flags to the corresponding classes (in this case the view) that conditionally expose functionality. In the simplest form this can be done by removing UI elements only and not touching any backend code.

feigelman commented on app/+QSPViewer/@Session/create.m:
```diff
@@ -123,13 +229,13 @@ function create(vObj)
-    'Parent', vObj.h.ParallelPanel,...
-    'Padding',5,...
-    'Spacing',10);
+% vObj.h.ParallelVbox = uix.VBox(...
```
delete commented lines

simkaryote replied:
In general UIX code should go away. The code is probably close to that.

feigelman commented on app/+QSPViewer/@Session/refresh.m:
```diff
@@ -40,6 +33,10 @@ function refresh(vObj)
     set(vObj.h.UseParallelCheckbox, 'Value', vObj.Data.UseParallel);
+    set(vObj.h.UseLogFile, 'Value', vObj.Data.UseLogging)
+    set(vObj.h.UseGit, 'Value', vObj.Data.AutoSaveGit)
+    set(vObj.h.UseSQL, 'Value', vObj.Data.UseSQL)
```
update/review

simkaryote replied:
I would prefer to not include new code in the refactor. In particular as this code is not written by MathWorks and it's going in with a pull request casting the code changes as MathWorks contributions. I am going to look the other way in this case.  But my recommendation going forward for your own sanity.

feigelman commented on app/+QSPViewer/@Session/update.m:
```diff
@@ -36,18 +28,25 @@ function update(vObj)
     RelativeAutoSavePath_new = '';
-    
+
+    UseLogging = true;
```
update/review

simkaryote replied:
same comment as above.

feigelman reviewed (commented):
generally we should remove unneeded .mat files, .xlsx files, old code from +QSPViewer.

Marked a few places where we need to clean up old vs new logger paradigm, and remove/update calls to store results in the SQLite DB. I did not carefully review the new classes in the QSPViewer, and instead focused on changes to the backend and other/utility code.

simkaryote commented:
I am gong to force the pull request into dev. It's been too long waiting for the approval and code changes from outside MathWorks are leaking into this pull request. I don't think it is controversial anymore that this new refactor will be part of v2.
